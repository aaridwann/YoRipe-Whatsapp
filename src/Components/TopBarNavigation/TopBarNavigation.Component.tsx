import React from 'react';
import { TouchableOpacity, View } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';
import * as Animatable from 'react-native-animatable';

import Constants from '../../Constants/Constants';
import { scaleSize } from '../../Utils/Size/size.utils';
import OptionsMenu from '../OptionsMenu/OptionsMenu.Component';
import { Props, TopBarHooksInterface } from './TopBarNavigation.Types';
import TopBarNavigationHooks from './TopBarNavigation.Hooks';
import SearchingBar from './SearchingBar/SearchingBar.Component';
import { styles } from './TopBarNavigation.Styles';

const _renderOptionsMenu = (hooks: TopBarHooksInterface, props: Props) => (
    <Animatable.View transition={'right'} style={styles.optionsMenuWrapper}>
        {hooks.state.isShowOptions && (
            <OptionsMenu
                isShow={hooks.state.isShowOptions}
                select={(val) => hooks.onSelectOptionHandler(val)}
                list={props.optionsList}
            />
        )}
    </Animatable.View>
);

const _renderBackIcon = (hooks: TopBarHooksInterface, props: Props) => (
    <TouchableOpacity onPress={props.onBack}>
        <Ionicons
            name="arrow-back-outline"
            size={scaleSize(25)}
            color={Constants.Colors.WHITE}
        />
    </TouchableOpacity>
);

const _renderOptionsIcon = (hooks: TopBarHooksInterface, props: Props) => (
    <React.Fragment>
        <TouchableOpacity
            style={styles.icon}
            onPress={() => hooks.showOptionsHandler()}
        >
            <Ionicons
                name="ellipsis-vertical"
                size={scaleSize(18)}
                color={Constants.Colors.WHITE}
            />
        </TouchableOpacity>
        {_renderOptionsMenu(hooks, props)}
    </React.Fragment>
);

const _renderRightSide = (hooks: TopBarHooksInterface, props: Props) => {
    return (
        <View style={styles.leftWrapper}>
            {props.rightHeaderComponent}
            {props.withOptions && _renderOptionsIcon(hooks, props)}
        </View>
    );
};

const _renderLeftSide = (hooks: TopBarHooksInterface, props: Props) => {
    return (
        <View style={styles.leftWrapper}>
            {props.backIcon && _renderBackIcon(hooks, props)}
            {props.leftHeaderComponent}
        </View>
    );
};

const _renderNavbarRegular = (props: Props, hooks: TopBarHooksInterface) => (
    <View style={styles.navbarRegularWrapper} accessibilityLabel="Navbar_Regular">
        {_renderLeftSide(hooks, props)}
        {_renderRightSide(hooks, props)}
    </View>
);

const _renderSearchingBar = (props: Props, hooks: TopBarHooksInterface) => (
    <SearchingBar
        isShow={props.isShowSearch}
        onSearch={props.inputSearchCallback}
        onBack={hooks.backHandler}
    />
);

const TopBarNavigation = (props: Props) => {
    const Hooks: TopBarHooksInterface = TopBarNavigationHooks(props);

    return (
        <View style={styles.container}>
            {props.isShowSearch
                ? _renderSearchingBar(props, Hooks)
                : _renderNavbarRegular(props, Hooks)}
        </View>
    );
};

export default TopBarNavigation;

TopBarNavigation.defaultProps = {
    withOptions: true,
    leftHeaderComponent: undefined,
    rightHeaderComponent: undefined,
    backIcon: false,
};
