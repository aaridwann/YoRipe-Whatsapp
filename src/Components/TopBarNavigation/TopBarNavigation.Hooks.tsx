import React from 'react';

import { Props, State, TopBarHooksInterface } from './TopBarNavigation.Types';

const TopBarNavigationHooks = (props: Props): TopBarHooksInterface => {
    const [state, setState] = React.useState<State>({
        isShowOptions: false,
        optionsSelected: '',
    });

    const showOptionsHandler = () =>
        setState(
            (prev): State => ({ ...prev, isShowOptions: !prev.isShowOptions })
        );

    const backHandler = () => {
        setState((prev) => ({
            ...prev,
            optionsSelected: '',
            isShowOptions: false,
        }));
        setTimeout(() => props.onBack(), 200);
    };

    const onProfileHandler = () => null;
    const onCallHandler = () => null;
    const onVideoHandler = () => null;
    const onSelectOptionHandler = (value: string) =>
        setState((prev): State => ({ ...prev, optionsSelected: value }));

    React.useEffect(() => {
        props.optionsSelected(state.optionsSelected);
    }, [state.optionsSelected]);

    return {
        state,
        backHandler,
        onProfileHandler,
        onCallHandler,
        onVideoHandler,
        onSelectOptionHandler,
        showOptionsHandler,
    };
};

export default TopBarNavigationHooks;
