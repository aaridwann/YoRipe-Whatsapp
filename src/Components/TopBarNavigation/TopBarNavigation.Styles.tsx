import { StyleSheet } from 'react-native';
import {
    scaleHeight,
    scaleSize,
    scaleWidth,
} from '../../Utils/Size/size.utils';
import Constants from '../../Constants/Constants';

export const styles = StyleSheet.create({
    container: {
        paddingHorizontal: scaleWidth(4),
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        width: '100%',
        height: scaleHeight(60),
        backgroundColor: Constants.Colors.GREEN,
        color: Constants.Colors.WHITE,
        zIndex: 2,
    },
    leftWrapper: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    avatar: {
        marginHorizontal: scaleWidth(4),
        width: scaleSize(30),
        height: scaleSize(30),
        borderRadius: scaleSize(30) / 2,
    },
    profile: {
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    },
    name: {
        color: Constants.Colors.WHITE,
        fontSize: scaleSize(15),
        fontWeight: '500',
    },
    lastSeen: {
        color: Constants.Colors.WHITE,
        fontSize: scaleSize(10),
        opacity: 0.5,
    },
    icon: {
        marginHorizontal: scaleSize(4),
    },
    optionsMenuWrapper: {
        position: 'absolute',
        top: scaleHeight(30),
        left: scaleWidth(-150),
    },
    navbarRegularWrapper: {
        alignItems: 'center',
        flexDirection: 'row',
        justifyContent: 'space-between',
        width: '100%',
    },
});
