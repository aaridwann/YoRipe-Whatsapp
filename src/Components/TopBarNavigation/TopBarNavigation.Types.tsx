export type Setter = (params?: string | number) => void;

export interface State {
    isShowOptions: boolean;
    optionsSelected: string;
}

export interface Props {
    leftHeaderComponent?: React.ReactNode;
    rightHeaderComponent?: React.ReactNode;
    optionsList: Array<string>;
    navigation?: (param: string) => void;
    optionsSelected: (menu: string) => void;
    isShowSearch: boolean;
    inputSearchCallback: (result: string) => void;
    onBack: VoidFunction;
    withOptions?: boolean;
    backIcon?: boolean;
}

export interface TopBarHooksInterface {
    state: State;
    backHandler: VoidFunction;
    onProfileHandler: Setter;
    onCallHandler: Setter;
    onVideoHandler: Setter;
    onSelectOptionHandler: (params: string) => void;
    showOptionsHandler: Setter;
}
