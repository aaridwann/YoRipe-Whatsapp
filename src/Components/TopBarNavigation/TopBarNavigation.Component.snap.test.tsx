import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import TopBarNavigation from './TopBarNavigation.Component';

const MockProps = {
    leftHeaderComponent: <></>,
    rightHeaderComponent: <></>,
    optionsList: ['setting'],
    navigation: jest.fn(),
    optionsSelected: jest.fn(),
    isShowSearch: false,
    inputSearchCallback: jest.fn(),
    onBack: jest.fn(),
    withOptions: true,
    backIcon: true,
};

const LeftHeaderComponent = () => <View accessibilityLabel="Left_Header" />;
const RightHeaderComponent = () => <View accessibilityLabel="Right_Header" />;

describe('TopBarNavigation', () => {
    const configs = [
        {
            desc: 'Without left and right header',
            props: MockProps,
        },
        {
            desc: 'With Only Left Header',
            props: {
                ...MockProps,
                leftHeaderComponent: LeftHeaderComponent(),
            },
        },
        {
            desc: 'With Only Right Header',
            props: {
                ...MockProps,
                rightHeaderComponent: RightHeaderComponent(),
            },
        },
        {
            desc: 'With Left and Right Header',
            props: {
                ...MockProps,
                leftHeaderComponent: LeftHeaderComponent(),
                rightHeaderComponent: RightHeaderComponent(),
            },
        },
        {
            desc: 'With Search true',
            props: {
                ...MockProps,
                isShowSearch: true,
                leftHeaderComponent: LeftHeaderComponent(),
                rightHeaderComponent: RightHeaderComponent(),
            },
        },
    ];

    configs.forEach(({ props, desc }) =>
        it(desc, () => {
            expect(render(<TopBarNavigation {...props} />)).toMatchSnapshot();
        })
    );
});
