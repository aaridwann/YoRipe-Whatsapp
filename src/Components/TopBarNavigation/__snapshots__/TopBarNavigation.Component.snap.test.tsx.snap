// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarNavigation With Left and Right Header 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#075e55",
      "color": "#ffffff",
      "flexDirection": "row",
      "height": 120.00000000000001,
      "justifyContent": "space-between",
      "paddingHorizontal": 8,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <View
    accessibilityLabel="Navbar_Regular"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        accessibilityLabel="Left_Header"
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityLabel="Right_Header"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        collapsable={false}
        style={
          {
            "left": -300,
            "position": "absolute",
            "right": 0,
            "top": 60.00000000000001,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TopBarNavigation With Only Left Header 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#075e55",
      "color": "#ffffff",
      "flexDirection": "row",
      "height": 120.00000000000001,
      "justifyContent": "space-between",
      "paddingHorizontal": 8,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <View
    accessibilityLabel="Navbar_Regular"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        accessibilityLabel="Left_Header"
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        collapsable={false}
        style={
          {
            "left": -300,
            "position": "absolute",
            "right": 0,
            "top": 60.00000000000001,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TopBarNavigation With Only Right Header 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#075e55",
      "color": "#ffffff",
      "flexDirection": "row",
      "height": 120.00000000000001,
      "justifyContent": "space-between",
      "paddingHorizontal": 8,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <View
    accessibilityLabel="Navbar_Regular"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityLabel="Right_Header"
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        collapsable={false}
        style={
          {
            "left": -300,
            "position": "absolute",
            "right": 0,
            "top": 60.00000000000001,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TopBarNavigation With Search true 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#075e55",
      "color": "#ffffff",
      "flexDirection": "row",
      "height": 120.00000000000001,
      "justifyContent": "space-between",
      "paddingHorizontal": 8,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <View
    accessibilityLabel="Navbar_Searching"
    collapsable={false}
    style={
      {
        "backgroundColor": "#054c44",
        "borderRadius": 20,
        "flexDirection": "row",
        "opacity": 0,
        "padding": 5,
        "transform": [
          {
            "translateY": -800,
          },
        ],
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityLabel="Navbar_Searching_Back_Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 16,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        style={
          {
            "borderRadius": 8,
            "minWidth": 520,
            "padding": 12,
          }
        }
      >
        <TextInput
          accessibilityLabel="Navbar_Searching_Input"
          onChangeText={[Function]}
          placeholder="Find a message"
          placeholderTextColor="white"
          style={
            {
              "color": "#ffffff",
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 16,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 16,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
    </View>
  </View>
</View>
`;

exports[`TopBarNavigation Without left and right header 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#075e55",
      "color": "#ffffff",
      "flexDirection": "row",
      "height": 120.00000000000001,
      "justifyContent": "space-between",
      "paddingHorizontal": 8,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <View
    accessibilityLabel="Navbar_Regular"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 8,
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
      <View
        collapsable={false}
        style={
          {
            "left": -300,
            "position": "absolute",
            "right": 0,
            "top": 60.00000000000001,
          }
        }
      />
    </View>
  </View>
</View>
`;
